src/util/records.py

```diff
@@ -410,8 +410,28 @@ def remove_records(cr, model, ids):
     table = table_of_model(cr, model)
     cr.execute('DELETE FROM "{}" WHERE id IN %s'.format(table), [ids])
     for ir in indirect_references(cr, bound_only=True):
-        query = 'DELETE FROM "{}" WHERE {} AND "{}" IN %s'.format(ir.table, ir.model_filter(), ir.res_id)
-        cr.execute(query, [model, ids])
+        if ir.company_dependent_comodel:
+            if ir.company_dependent_comodel == model:
+                cr.execute(
+                    f"""
+                    UPDATE {ir.table}
+                    SET {ir.res_id} = (
+                        SELECT jsonb_object_agg(
+                            key,
+                            CASE
+                                WHEN (value->>0)::int4 in %s THEN NULL
+                                ELSE value
+                            END)
+                        FROM jsonb_each({ir.res_id})
+                    )
+                    WHERE {ir.res_id} IS NOT NULL
+                    AND {ir.res_id} @? %s
```

Comment:
Can't we remove the key completely instead of setting null for the sake of simplicity?
nit: formatting
```suggestion
                    UPDATE {ir.table}
                       SET {ir.res_id} = (
                           SELECT jsonb_object_agg(key, value)
                             FROM jsonb_each({ir.res_id})
                            WHERE (value->>0)::int4 NOT IN %s
                           )
                     WHERE {ir.res_id} IS NOT NULL
                       AND {ir.res_id} @? %s
```

Reply:
Now with the details you gave about ir_defaults and how defaults work for the former ir_property, it's clear that we need to do as you do here. Thanks.